core/src/test/scala/unit/kafka/server/KafkaApisTest.scala

```diff
@@ -6222,4 +6222,62 @@ class KafkaApisTest {
 
     assertEquals(expectedResponse, response.data)
   }
+
+  @Test
+  def testGetTelemetrySubscriptionsNotAllowedForZkClusters(): Unit = {
+    val data = new GetTelemetrySubscriptionsRequestData()
+    data.setClientInstanceId(Uuid.randomUuid())
```

Comment:
Do we even need to set the client instance id? The request with no ID set should also get rejected? Should we test both cases?

Reply:
It won't make a difference to add or remove this as otherwise a ZERO_UUID will be sent. Leaving it as it is unless you suggest to remove it.

Reply:
I removed it though.